dev/PushNotifications/PushNotificationTelemetry.h

```diff
@@ -116,21 +118,61 @@ class PushNotificationTelemetry : public wil::TraceLoggingProvider
         return m_eventCount;
     }
 
-    wchar_t m_appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+    inline bool IsPackagedApp()
+    {
+        static bool isPackagedApp = AppModel::Identity::IsPackagedProcess();
+
+        return isPackagedApp;
+    }
+
+    inline const wchar_t* GetAppId()
+    {
+        static std::wstring appId = IsPackagedApp() ? InitAppIdPackaged() : InitAppIdUnpackaged();
+
+        return appId.c_str();
+    }
 
-    wchar_t* GetAppUserModelId()
+    std::wstring InitAppIdPackaged() noexcept
     {
-        if (m_appUserModelId[0] == '\0')
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+
+        UINT32 appUserModelIdSize = ARRAYSIZE(appUserModelId);
+        auto result = GetCurrentApplicationUserModelId(&appUserModelIdSize, appUserModelId);
+        if (result != ERROR_SUCCESS)
+        {
+            wcscpy_s(appUserModelId, L"AppUserModelId not found");
+            LOG_WIN32(result);
+        }
+
+        return appUserModelId;
+    }
+
+    PCWSTR CensorFilePath(_In_opt_ PCWSTR path) noexcept
+    {
+        if (path)
+        {
+            path = !PathIsFileSpecW(path) ? PathFindFileNameW(path) : path;
+        }
+
+        return path;
+    }
+
+    std::wstring InitAppIdUnpackaged() noexcept
+    {
+        std::wstring appId;
+
+        wil::unique_cotaskmem_string processName;
+        auto result = wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, processName);
+        if (result == ERROR_SUCCESS)
+        {
+            appId = CensorFilePath(processName.get());
```

Comment:
why are we storing the processname as appId? I think we need to call into the LRP and access the storage and get the appId right?

Reply:
It is just a field name. The GUID doesn't mean anything if not tied to a process name anyway.

Reply:
appId is not the best name choice; I'll replace it with appName. I was looking for an abstraction that would be the PFN for packaged app and the process name for unpackaged. AppId seemed like a good idea at the time but it's already way to overloaded.

Reply:
Changed to appName